### DISCUSSION TITLE
Vector Fitting Passivity Enforcements - Integration of Z/Y Parameters and General Discussion

### DISCUSSION THREAD

**Discussion by andree-sc** [scikit-rf/scikit-rf#851] [Ideas]
Continuing [this thread](https://groups.google.com/g/scikit-rf/c/VrX80hlDxx4), I would like to discuss the integration of passivity enforcement routines for Z and Y parameters. Other inputs (and problems) that might pop out along the way are also very welcome :)

**Comment by andree-sc** [comment]
I'm just gonna continue the thread from the google group here:

Yeah, so as told I've noticed some precision errors with the [method proposed]( https://ieeexplore.ieee.org/document/8880538), I don't entirely know which circumstances trigger it however...

I checked the code and source today and also noticed that [Gustavsens Method](https://ieeexplore.ieee.org/document/4512041) via the half-size test matrix is operating on symmetric networks... I think slight asymmetries are not that bad, but we should definitely keep that in the back of our heads. Unfortunately the methods that are valid for non-symmetric networks are a little more complicated.

Anyways, I think I'll try the algorithm proposed in [this](https://ieeexplore.ieee.org/document/8880538) (as mentioned in the earlier discussion) and it'll simply be limited to symmetrical networks for now.

Also: do we have examples for non-passive network (or rather networks that the VectorFitting class will produce a non-passive output for, given certain parameters)?

Cheers, Andree

**Reply by Vinc0110** [reply]
It's definitely fine to start with algorithms limited to symmetric networks. Again, the current implementation has the same limitation.

In terms of test data, have a look at the sNp files in `skrf/data` and `skrf/tests/`. Those are probably all S parameter files, but converting them to Z or Y should be ok for testing purposes. 

My all time favorites for test networks are:
- `skrf/data/ring_slot.s2p`, can also be imported via `nw = skrf.data.ring_slot`
- `skrf/tests/Agilent_E5071B.s4p`

And then I just found `skrf/tests/RS_ZNB8.s4p`, which I never used before, but appears to be (slightly) asymmetric.

For the [vector fitting example 2](https://scikit-rf.readthedocs.io/en/latest/examples/vectorfitting/vectorfitting_ex2_190ghz_active.html), I uploaded `doc/source/examples/vectorfitting/190ghz_tx_measured.S2P`, which is a measurement of an active 2-port.

As long as there is no conflict with copyright or licensing, you can also upload a few of your own files with Z or Y data.
